### PR TITLE
Implementing the quality IngestApi :)

### PR DESCRIPTION
**Pull request by Kjames5269**
See [Ion-Ingest-API](https://github.com/connexta/ion-ingest-api).

### CODE REVIEW

**Comment by Kjames5269** [comment]
build now

**Comment by Kjames5269** [comment]
❗️ ION-Ingest-API cannot be found on nexus.

**Comment by Kjames5269** [comment]
build now

File: ingest/src/main/java/com/connexta/ingest/endpoint/IngestController.java [hunk 1]
```diff
@@ -40,13 +36,16 @@ public IngestController(IngestService ingestService) {
...
-  @PostMapping(path = "/ingest", consumes = "application/json")
-  public ResponseEntity store(@RequestBody(required = false) JsonNode body) {
-    IngestRequest request = new IngestRequestImpl(body);
-    IngestResponse response = ingestService.ingest(request);
-    if (response.isStarted()) {
-      return new ResponseEntity(HttpStatus.OK);
-    }
-    return new ResponseEntity(HttpStatus.BAD_REQUEST);
+  //  Because ingest is a default method we need to override it
+  //  Thankfully however, Spring keeps the magic of annotations
```
Comment (emmberk) [inline ingest/src/main/java/com/connexta/ingest/endpoint/IngestController.java]: 👍

**Comment by emmberk** [comment]
I manually tested this. It should be good to merge once the CI build passes.

**Comment by emmberk** [comment]
build now

**Comment by emmberk** [comment]
```
Found 15 vulnerabilities in project ingest


One or more dependencies were identified with known vulnerabilities in ingest:

jackson-databind-nullable-0.1.0.jar (pkg:maven/org.openapitools/jackson-databind-nullable@0.1.0, cpe:2.3:a:fasterxml:jackson:0.1.0:*:*:*:*:*:*:*, cpe:2.3:a:fasterxml:jackson-databind:0.1.0:*:*:*:*:*:*:*) : CVE-2016-3720, CVE-2016-7051, CVE-2017-15095, CVE-2017-17485, CVE-2017-7525, CVE-2018-1000873, CVE-2018-14718, CVE-2018-5968, CVE-2018-7489
springfox-swagger2-2.9.2.jar (pkg:maven/io.springfox/springfox-swagger2@2.9.2, cpe:2.3:a:gradle:gradle:2.9.2:*:*:*:*:*:*:*) : CVE-2019-11065
springfox-swagger-common-2.9.2.jar (pkg:maven/io.springfox/springfox-swagger-common@2.9.2, cpe:2.3:a:gradle:gradle:2.9.2:*:*:*:*:*:*:*) : CVE-2019-11065
springfox-schema-2.9.2.jar (pkg:maven/io.springfox/springfox-schema@2.9.2, cpe:2.3:a:gradle:gradle:2.9.2:*:*:*:*:*:*:*) : CVE-2019-11065
springfox-spi-2.9.2.jar (pkg:maven/io.springfox/springfox-spi@2.9.2, cpe:2.3:a:gradle:gradle:2.9.2:*:*:*:*:*:*:*) : CVE-2019-11065
springfox-core-2.9.2.jar (pkg:maven/io.springfox/springfox-core@2.9.2, cpe:2.3:a:gradle:gradle:2.9.2:*:*:*:*:*:*:*) : CVE-2019-11065
guava-23.0.jar (pkg:maven/com.google.guava/guava@23.0, cpe:2.3:a:google:guava:23.0:*:*:*:*:*:*:*) : CVE-2018-10237


See the dependency-check report for more details.



> Task :ingest:dependencyCheckAnalyze FAILED
:ingest:dependencyCheckAnalyze (Thread[Execution worker for ':',5,main]) completed. Took 5.26 secs.

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':ingest:dependencyCheckAnalyze'.
> 

  Dependency-Analyze Failure:
  One or more dependencies were identified with vulnerabilities that have a CVSS score greater then '4.0': CVE-2017-15095, CVE-2017-17485, CVE-2018-1000873, CVE-2018-14718, CVE-2016-7051, CVE-2017-7525, CVE-2018-7489, CVE-2016-3720, CVE-2018-5968, CVE-2019-11065, CVE-2018-10237
  See the dependency-check report for more details.
```

File: build.gradle [hunk 2]
```diff
@@ -143,6 +138,7 @@ subprojects {
 	repositories {
 		mavenCentral()
 		mavenLocal()
+		maven { url = uri("http://nexus.phx.connexta.com:8081/nexus/content/groups/public/") }
```
Comment (jhunzik) [inline build.gradle]: ✏️ Nit: This limits the scanning to Connexta artifacts.
```suggestion
		maven { 
		     url = uri("http://nexus.phx.connexta.com:8081/nexus/content/groups/public/")
		     content {
		            includeGroup "com.connexta"
		      } 
		}
```

**Comment by ahoffer** [comment]
@Kjames5269 @emmberk This PR has been open for a week. I think we should, finish & merge it or close  it.

**Comment by emmberk** [comment]
I rebased and squashed commits so that it's easier to cherry-pick.

**Comment by emmberk** [comment]
I rebased and squashed commits so that it's easier to cherry-pick.

**Comment by emmberk** [comment]
https://github.com/connexta/ion-ingest-api/pull/6 was merged. This PR is no longer blocked.

**Comment by emmberk** [comment]
build now

**Comment by emmberk** [comment]
I'm overriding the CI build on this PR because Jenkins is being restarted.

**Pull request merged**